Formula/zstd.rb

```diff
@@ -26,6 +26,7 @@ def install
       system "cmake", "-S", ".", "-B", "builddir",
                       "-DZSTD_BUILD_CONTRIB=ON",
                       "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                      "-DZSTD_LEGACY_SUPPORT=1",
```

Comment:
Why isn't this the default? I don't think we want to deviate from what upstream thinks is a sensible default here.

Reply:
Per the README…

> By default, `zstd` supports decoding legacy formats >= v0.4.0 (`ZSTD_LEGACY_SUPPORT=4`).

Reply:
This doesn't appear to be true for CMake where `ZSTD_LEGACY_SUPPORT` is actually an `ON/OFF` boolean, defaulting to OFF. Turning it on implies level 5 rather than 4.

Reply:
facebook/zstd#3050 explains why that is.

@shadchin don't you just hate it when the documentation doesn't quite reflect the reality?

Reply:
Hello @carlocab @fanninpm @Bo98 Thank you for taking a look!

Indeed, as @Bo98 mentioned by default legacy format is disabled https://github.com/facebook/zstd/blob/b848c167abbb4c26a1f06a2b0187d9a70d7bc493/build/cmake/CMakeLists.txt#L87
Should I use `-DZSTD_LEGACY_SUPPORT=ON` instead? I'm not very familiar with `cmake` convention.

For my purpose, v0.5 is sufficient, I do not see for support of the format below this version.

Reply:
@fanninpm indeed it is possible to set arbitrary legacy support with `ZSTD_LEGACY_LEVEL`, should I set it to `1` to cover all versions?

Reply:
> Per the README…
> 
> > By default, `zstd` supports decoding legacy formats >= v0.4.0 (`ZSTD_LEGACY_SUPPORT=4`).

Sure. (Though `CMakeLists.txt` seems to contradict that.) That doesn't really explain why we shouldn't stick to what's the default in `CMakeLists.txt` -- presumably, there is a reason the default is configure that way. We should understand what that reason is before changing it to determine whether this change is appropriate.

Reply:
@carlocab This change is needed for my software that uses libzstd to decode binaries compressed with old ZSTD format

Reply:
I understand that you need it; you wouldn't have made this pull request if you didn't.

Still, that alone isn't enough to justify merging this in the absence of an understanding of the reason why this isn't the default behaviour.

For example, if this change causes performance regressions for users who don't need legacy support, then I don't think we can make this change to support a niche subset of users at the detriment of the many other users of this formula.

I suggest that you host the version of this formula that you need in your own tap for now.

Reply:
The only drawback is increased size of the library, no performance penalty.

OK, I will then use a formula from my own tap.

Reply:
What is the increase in size? Have you benchmarked (non-legacy) decoding with and without this change?

Reply:
There's slight increase of the library (less then 200KiB) since it supports more formats, no speed changes. NOTE, zstd benchmarks are flaky so you can get ~10% of drift. In the tests above I've run benchmark 4 times each and took one best and one worst result for each version of the build.
NOTE: both libs are built from source on my mac book.

With compatibility support:
```
*~>stat -c %s /usr/local/Cellar/zstd/1.5.2/lib/libzstd.1.5.2.dylib /usr/local/Cellar/zstd/1.5.2/lib/libzstd.a /usr/local/Cellar/zstd/1.5.2/bin/zstd
847768
961056
888976

*~>zstd -b
 3#Synthetic 50%     :  10000000 ->   3230847 (x3.095),  223.0 MB/s, 1925.4 MB/s
 3#Synthetic 50%     :  10000000 ->   3230847 (x3.095),  221.4 MB/s, 1886.9 MB/s
```

Without compatibility support:
```
*~>stat -c %s /usr/local/Cellar/zstd/1.5.2/lib/libzstd.1.5.2.dylib /usr/local/Cellar/zstd/1.5.2/lib/libzstd.a /usr/local/Cellar/zstd/1.5.2/bin/zstd
699280
763480
810368

*~>zstd -b
 3#Synthetic 50%     :  10000000 ->   3230847 (x3.095),  219.8 MB/s, 1925.5 MB/s
 3#Synthetic 50%     :  10000000 ->   3230847 (x3.095),  210.1 MB/s, 1822.3 MB/s
```

Reply:
@carlocab is there anything I can do to move forward? Do you prefer that backwards compatibility is enabled by default in the `zstd` cmake files instead? I don't see any consistency in the configuration of the zstd build: compatibility is enabled for the [VS](https://github.com/facebook/zstd/blob/e47e674cd09583ff0503f0f6defd6d23d8b718d3/build/VS2010/libzstd/libzstd.vcxproj#L184), [make](https://github.com/facebook/zstd/blob/c7e8315d886677e7d3fb81c56ff244fc1c808d7f/lib/libzstd.mk#L19) and [meson](https://github.com/facebook/zstd/blob/e47e674cd09583ff0503f0f6defd6d23d8b718d3/build/meson/meson_options.txt#L13) builds, and disabled for [cmake](https://github.com/facebook/zstd/blob/e47e674cd09583ff0503f0f6defd6d23d8b718d3/build/cmake/CMakeLists.txt#L85) and [single-file](https://github.com/facebook/zstd/blob/e47e674cd09583ff0503f0f6defd6d23d8b718d3/build/single_file_libs/zstd-in.c#L41) builds. I think single-file build is special, so it makes sense keep it disabled.

Reply:
Hi @niamster -- apologies for the delay here. (I have too many GitHub notifications.) I missed your last comment, but you did exactly what I would've suggested: open a pull request upstream that makes what you need the default.

I don't mind enabling this here now, and keeping it even when it's the default for the sake of being explicit.